### PR TITLE
Migrate code style to Black

### PR DESCRIPTION
**Pull request by alexandreroutier**
The following operations were performed (except DWI files due to PR #59):
- Apply `black` & `isort` to all files. `# fmt: on/off` for node connection and few other places for harmonization. More of them could be added in the future.
- Remove unused imports as much as possible
- Apply manually f-strings as much as possible. After 1st code review of @mdiazmel, I can say that we have an overall update
- Apply manually `pycodestyle` recommendations as much as possible (only "Missing Docstrings" category was ignored i.e. )
- For a few docstrings, convert them to Google Style (A PR should be dedicated for converters and some pipelines)
(ADNI files were modified after the merge of PR #163)

`utils/` folder is normally fully {f-string|pycodestyle|Google Style} compliant, I/O tools and converters should be tackled in the future. And pipelines if original author still contributes.

Documentation could be improved in terms of formatting & filling but this is beyond the initial goal of this PR.

### CODE REVIEW

**Review by mdiazmel — COMMENTED** [review]
Hi @alexandreroutier ,

Thanks for this nice PR!

I didn't check every single change induced by `Black`, I assume that it's doing its job. I found mostly some strings to reformat with f-strings and a couple of  `# ftm: on` to add. 

BTW, I didn't find where do you handle the parameters related to the style. I propose to include a file called `pyproject.toml` at the root folder with the following set up:
```
[tool.isort]
multi_line_output = 3
include_trailing_comma = true
force_grid_wrap = 0
use_parentheses = true
ensure_newline_before_comments = true
line_length = 88

[tool.black]
line-length = 88
target-version = ['py36', 'py37', 'py38']
include = '\.pyi?$'
exclude = '''
/(
    \.eggs
  | \.git
  | \.hg
  | \.mypy_cache
  | \.tox
  | \.venv
  | \.pytest_cache
  | _build
  | buck-out
  | build
  | dist
)/
'''
```
Also, edit the `.pep8speaks.yml` file with:

```
pycodestyle:
    ignore:
        - E203 # Ignore whitespace before ':'
        - E501 # Line too long
        - W503 # Line break occurred before a binary operator
```
Once all this is in place, we can iterate in another PR to add a pre-commit hook,

File: clinica/cmdline.py [hunk 1]
```diff
@@ -98,14 +111,26 @@ def custom_traceback(exc_type, exc_value, exc_traceback):
...
+        cprint(
+            f"\n{Fore.RESET}[Error] Program interrupted by the user. Clinica will now exit...{Fore.RESET}"
+        )
     else:
-        cprint(Fore.RED + '\n' + '*' * 23 + '\n*** Clinica crashed ***\n' + '*' * 23 + '\n' + Fore.RESET)
-        cprint('%sException type:%s %s' % (Fore.YELLOW, Fore.RESET, exc_type.__name__))
-        cprint('%sException value:%s %s' % (Fore.YELLOW, Fore.RESET, exc_value))
-        cprint('Below are displayed information that were gathered when Clinica crashed. This will help to understand '
-               'what happened if you transfer those information to the Clinica development team.\n')
+        cprint(
```
[the remark this answers is not in the document]
Comment (alexandreroutier) [reply clinica/cmdline.py:118]: After `black`, I manually applied f-string formatting. I missed several. This one will be fixed in another commit.

File: clinica/engine/cmdparser.py [hunk 2]
```diff
@@ -5,19 +5,18 @@
...
 from colorama import Fore
 
 PIPELINE_CATEGORIES = {
-    'CLINICA_COMPULSORY': '%sClinica mandatory arguments%s' % (Fore.BLUE, Fore.RESET),
-    'OPTIONAL': '%sPipeline options%s' % (Fore.BLUE, Fore.RESET),
-    'CLINICA_OPTIONAL': '%sClinica standard options%s' % (Fore.BLUE, Fore.RESET),
-    'ADVANCED': '%sPipelines advanced options%s' % (Fore.BLUE, Fore.RESET),
-    'IOTOOLS_OPTIONS': '%sOptional arguments%s' % (Fore.BLUE, Fore.RESET)
-    }
+    "CLINICA_COMPULSORY": "%sClinica mandatory arguments%s" % (Fore.BLUE, Fore.RESET),
```
[the remark this answers is not in the document]
Comment (alexandreroutier) [reply clinica/engine/cmdparser.py:14]: Will be done in another commit.

File: clinica/iotools/converters/adni_to_bids/adni_modalities/adni_flair.py [hunk 3]
```diff
@@ -26,16 +28,20 @@ def convert_adni_flair(source_dir, csv_dir, dest_dir, conversion_dir, subjs_list
...
+    )
+    images = compute_flair_paths(
+        source_dir, csv_dir, dest_dir, subjs_list, conversion_dir
+    )
+    cprint("Paths of FLAIR images found. Exporting images into BIDS ...")
     # flair_paths_to_bids(images, dest_dir)
-    paths_to_bids(images, dest_dir, 'flair', mod_to_update=mod_to_update)
-    cprint(Fore.GREEN + 'FLAIR conversion done.' + Fore.RESET)
+    paths_to_bids(images, dest_dir, "flair", mod_to_update=mod_to_update)
+    cprint(Fore.GREEN + "FLAIR conversion done." + Fore.RESET)
```
[the remark this answers is not in the document]
Comment (alexandreroutier) [reply clinica/iotools/converters/adni_to_bids/adni_modalities/adni_flair.py:44]: Yep! It is done for all `adni_<modality>.py` files

File: clinica/iotools/converters/adni_to_bids/adni_to_bids_cli.py [hunk 4]
```diff
@@ -52,16 +68,21 @@ def run_command(self, args):
 
         if args.clinical_data_only and args.force_new_extraction:
             raise ClinicaParserError(
-                '%s\n[Error] Arguments clinical_data_only and force_new_extraction are mutually exclusive.%s' %
-                (Fore.RED, Fore.RESET))
+                "%s\n[Error] Arguments clinical_data_only and force_new_extraction are mutually exclusive.%s"
```
[the remark this answers is not in the document]
Comment (alexandreroutier) [reply clinica/iotools/converters/adni_to_bids/adni_to_bids_cli.py:71]: Done.

File: clinica/iotools/converters/nifd_to_bids/nifd_to_bids_cli.py [hunk 5]
```diff
@@ -6,36 +6,45 @@
...
-        to_convert = convert_images(args.dataset_directory, args.bids_directory, args.clinical_data_directory)
-        convert_clinical_data(args.bids_directory, args.clinical_data_directory, to_convert)
-        cprint(Fore.GREEN + 'Conversion to BIDS succeeded' + Fore.RESET)
+        to_convert = convert_images(
+            args.dataset_directory, args.bids_directory, args.clinical_data_directory
+        )
+        convert_clinical_data(
+            args.bids_directory, args.clinical_data_directory, to_convert
+        )
+        cprint(Fore.GREEN + "Conversion to BIDS succeeded" + Fore.RESET)
```
[the remark this answers is not in the document]
Comment (alexandreroutier) [reply clinica/iotools/converters/nifd_to_bids/nifd_to_bids_cli.py:50]: Done.

File: clinica/pipelines/deeplearning_prepare_data/deeplearning_prepare_data_pipeline.py [hunk 6]
```diff
@@ -59,211 +58,259 @@ def build_input_node(self):
...
-                (save_as_pt, extract_slices, [('output_file', 'input_tensor')]),
-                (extract_slices, self.output_node, [('output_file_rgb', 'slices_rgb_T1')]),
-                (extract_slices, self.output_node, [('output_file_original', 'slices_original_T1')])
-                ])
-        elif self.parameters.get('extract_method') == 'patch':
-            self.connect([
-                (save_as_pt, extract_patches, [('output_file', 'input_tensor')]),
-                (extract_patches, self.output_node, [('output_patch', 'patches_T1')])
-                ])
+        # fmt: off
```
[the remark this answers is not in the document]
Comment (alexandreroutier) [reply clinica/pipelines/deeplearning_prepare_data/deeplearning_prepare_data_pipeline.py:289]: When I started applying `fmt: off` without closing with `on`, this was for ended files. I changed my mind in the meantime and agree with you.

**Pull request merged**